packages/shopify-api/docs/guides/oauth.md

```diff
@@ -1,22 +1,108 @@
 # Performing OAuth
 
+##### Table of content
+- [Supported types of OAuth Flow](#supported-types-of-oauth-flow)
+- [Token Exchange](#token-exchange)
+  - [Remix App](#remix-app)
+  - [Non-Remix App](#non-remix-app)
+  - [Detecting scope changes](#detecting-scope-changes)
+- [Authorization Code Grant Flow](#authorization-code-grant-flow)
+  - [Detecting scope changes](#detecting-scope-changes-1)
+- [After OAuth](#after-oauth)
+
+----------------------------------------------------------------------------------
 After you set up the library for your project, you'll be able to use it to interact with the APIs, and add your own functionality.
 The first thing your app will need to do is to get a token to access the Admin API by performing the OAuth process. Learn more about [OAuth on the Shopify platform](https://shopify.dev/docs/apps/auth/oauth).
 
-To perform OAuth, you will need to create two endpoints in your app:
+## Supported types of OAuth
+> [!TIP]
+> If you are building an embedded app, we **strongly** recommend using [Shopify managed install](ZL:TODO) with [Token Exchange](#token-exchange) instead of Authorizing Code Grant Flow.
 
-1. Start the process by calling [shopify.auth.begin](../reference/auth/begin.md) to redirect the merchant to Shopify, to ask for permission to install the app.
-1. Return the merchant to your app once they approve the app installation, by calling [shopify.auth.callback](../reference/auth/callback.md) to set up a session with an API access token.
+1. [Token Exchange](#token-exchange)
+    - Recommended for embedded apps
+    - Avoids redirect to your app to continue the OAuth flow
+    - Access scope changes are handled by Shopify if you use [Shopify managed install](ZL:TODO).
+2. [Authorization Code Grant Flow](#authorization-code-grant-flow)
+    - Suitable for non-embedded apps
+    - Installations, and access scope changes are managed by the app
 
-Once you complete the OAuth process, you'll be able to call [shopify.session.getCurrentId](../reference/session/getCurrentId.md) to fetch your session, and create API clients.
+## Token Exchange
+OAuth process by exchanging the current user's [Session Token](ZL:TODO) for an [Access Token](ZL:TODO).
 
-> **Note**: private apps are unable to perform OAuth, because they don't require an access token to interact with API.
+This can replace authorization code grant flow completely if you also take advantage of [Shopify managed install](ZL:TODO)
+
+To learn more about:
+  - [How Token Exchange Works](ZL:TODO)
+  - [Using Shopify Managed Install for your app's installation](ZL:TODO)
+  - [Configuring Access Scopes Through the Shopify CLI](https://shopify.dev/docs/apps/tools/cli/configuration)
+
+###### Remix App
```

Comment:
I think this is better suited in the shopify-app-remix README. This package is only an API package for JS devs so this info may not be as useful.

Reply:
Ah okay no worries! Makes sense 😄